### DISCUSSION TITLE
Using siliconcompiler with custom flow

### DISCUSSION THREAD

**Discussion by gretzteam** [siliconcompiler/siliconcompiler#902] [Development]
First let me say this is a fantastic project that I deeply hope will change the face of IC design in the future.

How much of silicon-compiler could be used as the framwework for a custom flow. I can see many semiconductor company benefiting from the python front-end but would need to use custom tools for backend flow (think the big 3 eda that open source tools cannot compete just now).

Firstly was the code base built with such a use case in mind where various class/functions could be overridden with custom code?

Second would this be something the community be interested in supporting. Of course NDA and proprietary code could not be shared back but probably a substantial amount could, benefiting everyone. 

Thanks again for the fantastic project. 

Dave

**Comment by aolofsson** [comment]
Thank you! The short answer is that SiliconCompiler already support a number of commercial EDA tools, but that code cannot be made public for NDA reasons. The same SiliconCompiler infrastruture can be used for proprietary/open source flows. Take a look at the openroad setup scripts. If you are familiar with commercial offerings, it should be clear how a similar driver could be created for proprietary tools.

https://github.com/siliconcompiler/siliconcompiler/blob/main/siliconcompiler/tools/openroad

**Reply by gretzteam** [reply]
Very interesting.
Regarding the support for comercial tools, is there any information about this? Is there a model in place for using this code (either through a purchase or licensing?)

Thanks!

**Reply by aolofsson** [reply]
It's complicated...If you are a current commercial eda customer, send me an email and let's disccuss what can be done. My email is andreas@zeroasic.com
